Front_End/templates/js/app.js

```diff
@@ -1,4 +1,187 @@
+    $urlRouterProvider.otherwise("/");
+
+    $stateProvider
+        .state('login', {
+          url: "/",
+          templateUrl: 'templates/html/login.html',
+          controller: "LoginCheckController"
+        })
+        .state('home', {
+          url: "/home",
+          templateUrl: 'templates/html/home.html',
+        })
+        .state('cart', {
+          url: "/cart",
+          templateUrl: 'templates/html/cart.html'
+        });
+});
+
+//Login Controller
+app.controller('LoginCheckController', ['$scope', '$location', LoginCheckController]);
+function LoginCheckController($scope, $location) {
+    /* Can be used for Admin login
+    $scope.users = [{
+        UserName: 'admin',
+        Password: 'password'
+    }];
+    */
+    $scope.LoginCheck = function() {
+        $location.path("home");
+    };
+}
+
+//Original Controller...May not be needed
 app.controller("catthings_ctrl", function ($scope, $window) {
- 
-}); 
\ No newline at end of file
+
+});
+
+//Service to manage cart
+app.service('cartList', [cartList]);
+function cartList () {
+  var cart = [];
+  return {
+      getCart: function () {
+          return cart;
+      },
+      addToCart: function(item) {
+          cart.push(item);
+          console.log(cart);
+      },
+      removeSelected: function(){
+        var newCart=[];
+            angular.forEach(cart,function(item){
+            if(!item.checked){
+                newCart.push(item);
+            }
+        });
+        cart=newCart;
+        return cart;
+      }
+  };
+}
+
+//NavBar Controller
+app.controller('NavBarController', ['$scope', NavBarController]);
+function NavBarController($scope) {
+    $scope.isCollapsed = true;
+}
+
+//Cart Controller
+app.controller('CartController', ['$scope', '$http', '$uibModal', '$location', 'cartList', CartController]);
+function CartController($scope, $http, $uibModal, $location, cartList){
+  $scope.cart = cartList.getCart();
+
+  function check(){
+    if(!$scope.cart.length){
+      $scope.cartNotEmpty = false;
+      $scope.cartEmpty = true;
+    }
+    else{
+      $scope.cartNotEmpty = true;
+      $scope.cartEmpty = false;
+    }
+  }
+  check()
+
+  //Remove selected items from cart
+  $scope.removeSelected = function() {
+    $scope.cart = cartList.removeSelected();
+    check();
+  }
+
+  //Checkout
+  $scope.checkOut = function() {
+    console.log($scope.cart);
+
+    $http.post('/checkout', $scope.cart).then(function(response){
+      console.log(response.status);
+      if(response.status === 200){
+              $location.path("home");
+      }
+      //Else 404 error....Could need another modal
+    });
+  }
+}
+
+//Inventory Controller
+app.controller('InventoryController', ['$scope', '$http', '$uibModal', '$location', 'cartList', InventoryController]);
+function InventoryController($scope, $http, $uibModal, $location, cartList) {
+  //Get latest inventory data from database
+  $http.get('/view').then(function (response) {
```

Comment:
Lets get this pointing to the API before we merge.  http://things.cs.pdx.edu:3000/items